### PR TITLE
Localization: add other ndt map files for pyramid_map module

### PR DESCRIPTION
**Pull request by tthhee**

### CODE REVIEW

File: modules/localization/msf/local_pyramid_map/ndt_map/ndt_map_node_config.cc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+namespace apollo {
+namespace localization {
+namespace msf {
+
+NdtMapNodeConfig::NdtMapNodeConfig() {}
+
+NdtMapNodeConfig::~NdtMapNodeConfig() {}
+
+BaseMapNodeConfig* NdtMapNodeConfig::Clone() {
+  NdtMapNodeConfig* map_node_config = new NdtMapNodeConfig();
```
Comment (jinghaomiao) [inline modules/localization/msf/local_pyramid_map/ndt_map/ndt_map_node_config.cc]: Use shared pointer

File: modules/localization/msf/local_pyramid_map/ndt_map/ndt_map_node.cc [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  map_node_config_->map_version_ = map_config_->GetMapVersion();
+  map_node_config_->has_map_version_ = false;
+  map_node_config_->has_body_md5_ = false;
+  is_reserved_ = false;
+  data_is_ready_ = false;
+  is_changed_ = false;
+  num_valid_cells_ = 0;
+  num_valid_single_cells_ = 0;
+
+  map_matrix_ = new NdtMapMatrix();
```
Comment (anyong298) [inline modules/localization/msf/local_pyramid_map/ndt_map/ndt_map_node.cc]: smarter_pointer is preferred. This applies to other pointers too.

Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/ndt_map/ndt_map_node.cc]: fixed

File: modules/localization/msf/local_pyramid_map/ndt_map/ndt_map_node_config.cc (hunk 1 above)
Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/ndt_map/ndt_map_node_config.cc]: fixed

**Pull request merged**